src/meta/api/src/background_api_impl.rs

```diff
@@ -0,0 +1,363 @@
+    }
+
+    async fn update_background_job(
+        &self,
+        req: UpdateBackgroundJobReq,
+    ) -> Result<UpdateBackgroundJobReply, KVAppError> {
+        debug!(req = debug(&req), "BackgroundApi: {}", func_name!());
+        let name_key = &req.job_name;
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+
+        loop {
+            trials.next().unwrap()?;
+            let (seq, id) = get_u64_value(self, name_key).await?;
+            debug!(seq, id, ?name_key, "update_background_job");
+            let id_key = match seq {
+                0 => {
+                    let id = fetch_id(self, IdGenerator::background_job_id()).await?;
+                    BackgroundJobId { id }
+                }
+                _ => BackgroundJobId { id },
+            };
+            let (id_name_seq, _): (_, Option<BackgroundJobInfo>) =
+                get_pb_value(self, &id_key).await?;
+            let meta = req.info.clone();
+            let condition = vec![txn_cond_seq(&id_key, Eq, id_name_seq)];
+            let if_then = vec![
+                txn_op_put(name_key, serialize_u64(id_key.id)?),
+                txn_op_put(&id_key, serialize_struct(&meta)?),
+            ];
+            let txn_req = TxnRequest {
+                condition,
+                if_then,
+                else_then: vec![],
+            };
+            let (succ, _responses) = send_txn(self, txn_req).await?;
+            if succ {
+                break Ok(UpdateBackgroundJobReply { id: id_key.id });
+            }
+        }
+    }
+
+    async fn update_background_job_status(
+        &self,
+        req: UpdateBackgroundJobStatusReq,
+    ) -> Result<UpdateBackgroundJobReply, KVAppError> {
+        let name = &req.job_name;
+        let (id, mut resp) =
+            get_background_job_or_error(self, name, "failed to get background job").await?;
+        if resp.job_status == Some(req.status.clone()) {
+            return Ok(UpdateBackgroundJobReply { id });
+        }
+        resp.job_status = Some(req.status);
+        resp.last_updated = Some(Utc::now());
+        let req = UpdateBackgroundJobReq {
+            job_name: name.clone(),
+            info: resp,
+        };
+        self.update_background_job(req).await
+    }
+
+    async fn update_background_job_params(
+        &self,
+        req: UpdateBackgroundJobParamsReq,
+    ) -> Result<UpdateBackgroundJobReply, KVAppError> {
+        let name = &req.job_name;
+        let (id, mut resp) =
+            get_background_job_or_error(self, name, "failed to get background job").await?;
+        if resp.job_params == Some(req.params.clone()) {
+            return Ok(UpdateBackgroundJobReply { id });
+        }
+        resp.job_params = Some(req.params);
+        resp.last_updated = Some(Utc::now());
+        let req = UpdateBackgroundJobReq {
+            job_name: name.clone(),
+            info: resp,
+        };
+        self.update_background_job(req).await
```

Comment:
Is this a non-conditional update? i.e., the state after an update does not have to depend on the state before the update.

Implementing update in this way will get an update lost:
```
initial state    process B           process C
A              
|
|    read
+-------------------------------------> A
|                                       | local update
|    read                               | line 220-225
+----------------> A                    |
|                  | local update       | 
|                  | line 220-225       |
|    write         v                    |
B <--------------- B                    |
|                                       |
|                                       |
|                            write      v
C <-------------------------------------C
```

If this is acceptable, there is no need to use transaction. The simplified way is:

- Load the `name->id` record and the `id->value` record.

- Update the `id->value` record with the new value using the `upsert_kv()`. Set the `UpsertKV.seq` field to `MatchSeq::GE(0)` to update any existing record with a sequence number greater than or equal to 0. This will not create a new record if the `id->value` does not exist.

Such a way is feasible because renaming does not affect the `id->value` record, but only affect the `name->id` record. Therefore we only need to forbid a delete operation during updating, with `MatchSeq::GE(0)`.